### DISCUSSION TITLE
How to specify inputs for a certain package?

### DISCUSSION THREAD

**Discussion by Sczlog** [vercel/turborepo#3857] [Help]
In my case, I have monorepo with 3 package: server, ui and utils.
All the package have generate pipeline to generate some code by plugin, and they have (and they can only have) a common inputs as `config/**` as their codegen config.
In server package I have a `schema.graphql` as base schema, and ui package need the schema to run generate pipeline, so I add `../server/schema.graphql` to inputs.
But turbo don't support specify input for a certain package, and in utils package, generate pipeline do not need `schema.graphql` as inputs, so if this file change, utils package will also run generate, although it has nothing new.

My temporary solution is try to get all required files for single package with globby and calculate their hash, then set the result as an input file, but some of my collegues think it is weird as turbo has already calculated hash.
Another try it to use ui:generate, utils:generate but I think it will intro too much different generate pipeline which will mess up in the end.
So if there is some way can set different inputs for specified package in the same pipeline?

**Comment by anthonyshew** [comment]
You can set inputs using the `inputs` key: https://turbo.build/repo/docs/reference/configuration#inputs

**Reply by mistic100** [reply]
That does not answer the question which about setting **different** inputs for a specific package.

The anwser is here https://turbo.build/repo/docs/crafting-your-repository/configuring-tasks#depending-on-a-specific-task-in-a-specific-package

Example based on original question:

```js
{
  "tasks": {
    "build": { // applied to utils and ui
      "inputs": ["config/**"] 
    },
    "server#build": { // applied to server
      "inputs": ["config/**", "schema.graphql"]
    }
  }
}
```
